nixos/modules/programs/looking-glass.nix

```
@@ -0,0 +1,401 @@
+          type = types.nullOr types.bool;
+          description = "Scale cursor input position to screen size when up/down scaled";
+          default = null;
+        };
+        captureOnStart = mkOption {
+          type = types.nullOr types.bool;
+          description = "Capture mouse and keyboard on start";
+          default = null;
+        };
+        alwaysShowCursor = mkOption {
+          type = types.nullOr types.bool;
+          description = "Always show host cursor";
+          default = null;
+        };
+        showCursorDot = mkOption {
+          type = types.nullOr types.bool;
+          description = "Use a “dot” cursor when the window does not have focus";
+          default = null;
+        };
+      };
+      egl = {
+        vsync = mkOption {
+          type = types.nullOr types.bool;
+          description = "Enable vsync";
+          default = null;
+        };
+        doubleBuffer = mkOption {
+          type = types.nullOr types.bool;
+          description = "Enable double buffering";
+          default = null;
+        };
+        multisample = mkOption {
+          type = types.nullOr types.str;
+          description = "Enable Multisampling";
+          default = null;
+        };
+        nvGainMax = mkOption {
+          type = types.nullOr types.int;
+          description = "The maximum night vision gain";
+          default = null;
+        };
+        nvGain = mkOption {
+          type = types.nullOr types.int;
+          description = "The initial night vision gain at startup";
+          default = null;
+        };
+        cbMode = mkOption {
+          type = types.nullOr types.int;
+          description = "Color Blind Mode (0 = Off, 1 = Protanope, 2 = Deuteranope, 3 = Tritanope)";
+          default = null;
+        };
+        scale = mkOption {
+          type = types.nullOr types.int;
+          description = "Set the scale algorithm (0 = auto, 1 = nearest, 2 = linear)";
+          default = null;
+        };
+      };
+      opengl = {
+        mipmap = mkOption {
+          type = types.nullOr types.bool;
+          description = "Enable mipmapping";
+          default = null;
+        };
+        vsync = mkOption {
+          type = types.nullOr types.bool;
+          description = "Enable vsync";
+          default = null;
+        };
+        preventBuffer = mkOption {
+          type = types.nullOr types.bool;
+          description = "Prevent the driver from buffering frames";
+          default = null;
+        };
+        amdPinnedMem = mkOption {
+          type = types.nullOr types.bool;
+          description = "Use GL_AMD_pinned_memory if it is available";
+          default = null;
+        };
+      };
+      wayland = {
+        warpSupport = mkOption {
+          type = types.nullOr types.bool;
+          description = "Enable cursor warping";
+          default = null;
+        };
+      };
+    };
+
+    extraConfig = mkOption {
+      type = types.str;
```

```suggestion
      type = types.lines;
```

`types.str` does not support merging multiple definitions whereas `types.lines` does.

---

Decided to just remove `extraConfig` because it doesn't look like it's ever done in this kind of situation.